spec/index.bs

```diff
@@ -3693,21 +3693,111 @@ GPURenderPipeline includes GPUPipelineBase;
 
 <script type=idl>
 dictionary GPURenderPipelineDescriptor : GPUPipelineDescriptorBase {
+    GPUVertexStateDescriptor vertexState = {};
     required GPUProgrammableStageDescriptor vertexStage;
-    GPUProgrammableStageDescriptor fragmentStage;
-
     required GPUPrimitiveTopology primitiveTopology;
     GPURasterizationStateDescriptor rasterizationState = {};
-    required sequence<GPUColorStateDescriptor> colorStates;
+    GPUProgrammableStageDescriptor fragmentStage;
     GPUDepthStencilStateDescriptor depthStencilState;
-    GPUVertexStateDescriptor vertexState = {};
-
-    GPUSize32 sampleCount = 1;
-    GPUSampleMask sampleMask = 0xFFFFFFFF;
-    boolean alphaToCoverageEnabled = false;
+    required sequence<GPUColorStateDescriptor> colorStates;
 };
 </script>
 
+A {{GPURenderPipelineDescriptor}} describes the state of a render [=pipeline=] by
+configuring each of the [=render stages=] in the order they apply.
+
+<div algorithm>
+    <dfn abstract-op>render</dfn>(descriptor, drawCall, indexBuffer, vertexBuffers, bindGroups)
+        **Arguments:**
+            - |descriptor|: Description of the current {{GPURenderPipeline}}.
+            - |drawCall|: The draw call parameters.
+            - |indexBuffer|: Currently bound index buffer via {{GPURenderEncoderBase/setIndexBuffer()}}.
+            - |vertexBuffers|: [=list=]&lt;vertex buffer&gt; bound by {{GPURenderEncoderBase/setVertexBuffer()}}.
+            - |bindGroups|: [=list=]&lt;{{GPUBindGroup}}&gt; bound by {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}}.
+
+        Issue: define somewhere what a render pass state is?
+        Let |vertexState| be |descriptor|.{{GPURenderPipelineDescriptor/vertexState}}.
+
+        1. Resolve indices.
+            The pipeline builds a list of vertices to process for each instance.
+            If |vertexState|.{{GPUVertexStateDescriptor/indexFormat}} is not `null`:
```

Comment:
I think we want to check if the drawcall is indexed or not here instead of the indexFormat

Reply:
good point!